### PR TITLE
JVM mode builds

### PR DESCRIPTION
**Pull request by matejvasek**

### CODE REVIEW

**Comment by matejvasek** [comment]
@lance this is draft but pleas give it short look.

File: Makefile [hunk 1]
```diff
@@ -7,13 +7,15 @@ BASE_REPO  := quay.io/boson/faas-stack-base
 BUILD_REPO := quay.io/boson/faas-stack-build
 RUN_REPO   := quay.io/boson/faas-stack-run
 
-QUARKUS_BUILDER_REPO := quay.io/boson/faas-quarkus-builder
+QUARKUS_NATIVE_BUILDER_REPO := quay.io/boson/faas-quarkus-native-builder
```
Comment (zroubalik) [inline Makefile:10]: very minor nit picking: I'd love to see this sorted, eg the QUARKUS_* variables next to each other (the same for the rest of the Makefile)

File: Makefile [hunk 2]
```diff
@@ -23,34 +25,38 @@ stacks:
 	./stacks/build-stack.sh -v $(VERSION_TAG) stacks/ubi8
 	./stacks/build-stack.sh -v $(VERSION_TAG) stacks/ubi8-minimal
 	./stacks/build-stack.sh -v $(VERSION_TAG) stacks/nodejs
-	./stacks/build-stack.sh -v $(VERSION_TAG) stacks/quarkus
+	./stacks/build-stack.sh -v $(VERSION_TAG) stacks/quarkus-native
```
Comment (zroubalik) [inline Makefile:28]: dtto

File: stacks/quarkus-jvm/build/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
+ARG version=tip
+FROM quay.io/boson/faas-stack-build:ubi8-${version}
+
+ARG stack_id
+ENV CNB_STACK_ID=${stack_id}
+LABEL io.buildpacks.stack.id=${stack_id}
+
+USER root
+RUN dnf module install -y maven:3.6
```
Comment (zroubalik) [inline stacks/quarkus-jvm/build/Dockerfile:9]: Can we merge these two RUNs? So we get rid of the extra layer.

File: stacks/quarkus-jvm/build/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+RUN dnf install -y wget maven tar gzip java-11-openjdk-headless \
+    && dnf update -y \
+    && dnf clean all -y
+
+ENV HOME /projects/quarkus-function
+WORKDIR $HOME
+RUN chown cnb:cnb $HOME
```
Comment (zroubalik) [inline stacks/quarkus-jvm/build/Dockerfile:16]: Could this be included in the RUN above?

File: stacks/quarkus-jvm/run/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    && microdnf update \
+    && microdnf clean all \
+    && curl https://repo1.maven.org/maven2/io/fabric8/run-java-sh/${RUN_JAVA_VERSION}/run-java-sh-${RUN_JAVA_VERSION}-sh.sh -o /usr/local/bin/run-java.sh \
+    && chmod "a+rx" /usr/local/bin/run-java.sh \
+    && echo "securerandom.source=file:/dev/urandom" >> /etc/alternatives/jre/lib/security/java.security
+
+# Configure the JAVA_OPTIONS, you can add -XshowSettings:vm to also display the heap size.
+ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
+
+RUN chown cnb:cnb $HOME
```
Comment (zroubalik) [inline stacks/quarkus-jvm/run/Dockerfile:29]: Could this be included in the RUN above?

**Review by zroubalik — COMMENTED** [review]
minor nits

**Review by lance — COMMENTED** [review]
Looks good to me overall @matejvasek. The comments from @zroubalik are good suggestions, but iirc these changes could be applied to all of the Dockerfiles.

**Review by lance — APPROVED** [review]

**Pull request merged**